Framework/Algorithms/src/NormaliseToMonitor.cpp

```diff
@@ -561,99 +662,123 @@ void NormaliseToMonitor::normaliseBinByBin(
-
-    if (!m_commonBins) {
-      // ConvertUnits can give X vectors of all zeros - skip these, they cause
-      // problems
-      if (X.back() == 0.0 && X.front() == 0.0)
-        continue;
-      // Rebin the monitor spectrum to match the binning of the current data
-      // spectrum
-      VectorHelper::rebinHistogram(
-          monX.rawData(), monY.mutableRawData(), monE.mutableRawData(),
-          X.rawData(), Y.mutableRawData(), E.mutableRawData(), false);
-      // Recalculate the overall normalization factor
-      this->normalisationFactor(X, Y, E);
-    }
+  std::shared_ptr<SpectrumInfo> inputSpecInfo;
+  std::shared_ptr<SpectrumInfo> monitorSpecInfo;
+  if (m_syncScanInput) {
+    inputSpecInfo =
+        std::make_shared<SpectrumInfo>(inputWorkspace->spectrumInfo());
+    monitorSpecInfo = std::make_shared<SpectrumInfo>(m_monitor->spectrumInfo());
+  }
 
-    if (inputEvent) {
-      // --- EventWorkspace ---
-      EventList &outEL = outputEvent->getSpectrum(i);
-      outEL.divide(X.rawData(), Y.mutableRawData(), E.mutableRawData());
-    } else {
-      // --- Workspace2D ---
-      auto &YOut = outputWorkspace->mutableY(i);
-      auto &EOut = outputWorkspace->mutableE(i);
-      const auto &inY = inputWorkspace->y(i);
-      const auto &inE = inputWorkspace->e(i);
-      outputWorkspace->mutableX(i) = inputWorkspace->x(i);
-
-      // The code below comes more or less straight out of Divide.cpp
-      for (size_t k = 0; k < specLength; ++k) {
-        // Get the input Y's
-        const double leftY = inY[k];
-        const double rightY = Y[k];
-
-        if (rightY == 0.0) {
-          hasZeroDivision = true;
-        }
-
-        // Calculate result and store in local variable to avoid overwriting
-        // original data if output workspace is same as one of the input ones
-        const double newY = leftY / rightY;
-
-        if (fabs(rightY) > 1.0e-12 && fabs(newY) > 1.0e-12) {
-          const double lhsFactor = (inE[k] < 1.0e-12 || fabs(leftY) < 1.0e-12)
-                                       ? 0.0
-                                       : pow((inE[k] / leftY), 2);
-          const double rhsFactor =
-              E[k] < 1.0e-12 ? 0.0 : pow((E[k] / rightY), 2);
-          EOut[k] = std::abs(newY) * sqrt(lhsFactor + rhsFactor);
-        }
-
-        // Now store the result
-        YOut[k] = newY;
-      } // end Workspace2D case
-    }   // end loop over current spectrum
-
-    PARALLEL_END_INTERUPT_REGION
-  } // end loop over spectra
-  PARALLEL_CHECK_INTERUPT_REGION
-  if (hasZeroDivision) {
-    g_log.warning() << "Division by zero in some of the bins.\n";
+  for (auto &workspaceIndex : m_workspaceIndexes) {
+    // Get hold of the monitor spectrum
+    const auto &monX = m_monitor->binEdges(workspaceIndex);
+    auto monY = m_monitor->counts(workspaceIndex);
+    auto monE = m_monitor->countStandardDeviations(workspaceIndex);
+    size_t timeIndex = 0;
+    if (m_syncScanInput)
```

Comment:
Can remove `if`?

Reply:
This `if` is required to avoid a crash in the unit tests when getting the spectrum definitions for a workspace without and instrument.

Reply:
I see, I guess the problem is that no detector is associated to the monitor so the spectrum definition is empty.